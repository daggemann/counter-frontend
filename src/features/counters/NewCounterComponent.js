import React from 'react';
import styled from 'styled-components';

const NewCounterComponent = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    ${props => (props.width) ? 'width: ' + props.width : ''};
    ${props => (props.height) ? 'height: ' + props.height : ''};
`;

const Input = styled.input`
    border-top: 0px solid gray;
    border-right: 0px solid gray;
    border-bottom: 1px solid gray;
    border-left: 0px solid gray;
`;

const AddButton = styled.button`
    border: 1px solid gray;
    border-radius: 1em 1em 1em 1em;
    width: 100px;
    
    &:hover {
        background-color: #2ECC40;
    }
`;

export default class NewCounterContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            title: ''
        };
        this.handleChange = this.handleChange.bind(this);
        this.addCounter = this.addCounter.bind(this);
    }

    handleChange(e) {
        this.setState({title: e.target.value});
    }

    addCounter() {
        if (this.state.title.length === 0) {
            return
        }
        this.props.onClick(this.state.title);
        this.setState({title: ''})
    }

    render() {
        return (
            <NewCounterComponent width={'300px'} height={'30px'}>
                <Input
                    type='text'
                    placeholder='Counter title'
                    value={this.state.title}
                    onChange={this.handleChange}
                    size={15}
                />
                <AddButton onClick={this.addCounter}>Add</AddButton>
            </NewCounterComponent>
        )
    }
}