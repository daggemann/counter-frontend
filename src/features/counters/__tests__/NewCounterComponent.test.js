import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import NewCounterComponent from '../NewCounterComponent'

test('render new counter', () => {
    const tree = renderer.create(<NewCounterComponent/>);
    expect(tree.toJSON()).toMatchSnapshot();

    const instance = tree.getInstance();
    instance.setState({title: 'test'});

    expect(tree.toJSON()).toMatchSnapshot();
});

test('ensure that title state is given in input', () => {
    const wrapper = shallow(<NewCounterComponent/>);
    let input = wrapper.find('Input');
    expect(input.get(0).props.value).toEqual('');

    wrapper.setState({title: 'test'});
    input = wrapper.find('Input');
    expect(input.get(0).props.value).toEqual('test');
});

test('ensure that we are able to add counter when title state is non-blank', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(<NewCounterComponent onClick={onClickMock}/>);
    wrapper.setState({title: 'test'});
    const button = wrapper.find('AddButton');
    button.simulate('click')
    expect(onClickMock).toHaveBeenCalledTimes(1);
});

test('ensure that we are not able to add counter when title state is blank', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(<NewCounterComponent onClick={onClickMock}/>);
    const button = wrapper.find('AddButton');
    button.simulate('click')
    expect(onClickMock).toHaveBeenCalledTimes(0);
});
