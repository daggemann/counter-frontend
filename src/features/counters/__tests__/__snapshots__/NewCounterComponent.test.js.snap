// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render new counter 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 300px;
  height: 30px;
}

.c1 {
  border-top: 0px solid gray;
  border-right: 0px solid gray;
  border-bottom: 1px solid gray;
  border-left: 0px solid gray;
}

.c2 {
  border: 1px solid gray;
  border-radius: 1em 1em 1em 1em;
  width: 100px;
}

.c2:hover {
  background-color: #2ECC40;
}

<div
  className="c0"
  height="30px"
  width="300px"
>
  <input
    className="c1"
    onChange={[Function]}
    placeholder="Counter title"
    size={15}
    type="text"
    value=""
  />
  <button
    className="c2"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`render new counter 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 300px;
  height: 30px;
}

.c1 {
  border-top: 0px solid gray;
  border-right: 0px solid gray;
  border-bottom: 1px solid gray;
  border-left: 0px solid gray;
}

.c2 {
  border: 1px solid gray;
  border-radius: 1em 1em 1em 1em;
  width: 100px;
}

.c2:hover {
  background-color: #2ECC40;
}

<div
  className="c0"
  height="30px"
  width="300px"
>
  <input
    className="c1"
    onChange={[Function]}
    placeholder="Counter title"
    size={15}
    type="text"
    value="test"
  />
  <button
    className="c2"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
